src/chunk_constraint.c

```diff
@@ -341,106 +341,55 @@ ts_chunk_constraints_create(ChunkConstraints *ccs, Oid chunk_oid, int32 chunk_id
 								hypertable_id);
 }
 
-/*
- * Scan filter function for only getting dimension constraints.
- */
-static ScanFilterResult
-chunk_constraint_for_dimension_slice(TupleInfo *ti, void *data)
-{
-	if (heap_attisnull_compat(ti->tuple, Anum_chunk_constraint_dimension_slice_id, ti->desc))
-		return SCAN_EXCLUDE;
-
-	return SCAN_INCLUDE;
-}
-
-static ScanTupleResult
-chunk_constraint_tuple_found(TupleInfo *ti, void *data)
+static void
+scan_by_chunk_id(ScannerCtx *ctx, int chunk_id)
```

Comment:
I find the names of these `scan_by_` functions confusing because they imply that they are performing some task (scanning) while they are actually just initializing a struct.

I think they should be called something like `init_scan_by_chunk_id`, `scan_by_chunk_id_init` or something. Also, aren't we generally using `int32` and not `int` for ID types?

Reply:
changed to the init_ prefix